### PR TITLE
Fix py2 compatibility for SelfRepairingSession

### PR DESCRIPTION
**Pull request by vereszol**

### CODE REVIEW

File: src/crl/interactivesessions/ShellSubprocess.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def _run(self):
 
     def _backup(self, python_object):
         with open(self.pickled_backup, 'w') as backup:
-            pickle.dump(python_object, backup, protocol=0)
+            pickle.dump(python_object, backup, protocol=2)
```
Comment (petrieh) [inline src/crl/interactivesessions/ShellSubprocess.py]: protocol=2 does not work with Python2

File: src/crl/interactivesessions/ShellSubprocess.py [hunk 2]
```diff
@@ -110,7 +110,7 @@ def get_start_trigger(self):
 
     @staticmethod
     def _serialize(python_object):
-        return base64.b64encode(pickle.dumps(python_object, protocol=0))
+        return base64.b64encode(pickle.dumps(python_object, protocol=2))
```
Comment (petrieh) [inline src/crl/interactivesessions/ShellSubprocess.py]: protocol 2 is not cross-platform compatible: Python 2 protocol 2 is not compatible with Python 3 protocol 2.

File: src/crl/interactivesessions/ShellSubprocess.py (hunk 1 above)
Comment (vereszol) [reply src/crl/interactivesessions/ShellSubprocess.py]: I'll revert to protocol=0.

File: src/crl/interactivesessions/ShellSubprocess.py (hunk 2 above)
Comment (vereszol) [reply src/crl/interactivesessions/ShellSubprocess.py]: I'll revert to protocol=0.

File: .pylintrc [hunk 3]
```diff
@@ -69,7 +69,8 @@ disable=missing-docstring,anomalous-backslash-in-string,
     relative-import,redefined-outer-name,redefined-builtin,eval-used,
     exec-used,wildcard-import,
     inconsistent-return-statements,fixme,no-name-in-module,abstract-method,
-    useless-object-inheritance
+    useless-object-inheritance,no-else-break,unnecessary-comprehension,
+    invalid-overridden-method,invalid-overridden-method
```
Comment (petrieh) [inline .pylintrc]: typo: invalid-overriden-method is disabled twice

File: robottests/ClusterExec.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                                         work_dir=work_dir,
+                                         management_node_floating_ip=self.host_name)
+        return ise
+
+    def run_cmd_in_node(self, node, cmd, timeout=120, validate_return_status=False):
+        """
+        Run command in target node
+
+        Example:
+        | ${status} |  ${stdout} |  ${stderr}= | Run Cmd In Node |  CLA-0 | "echo Hello" | timeout=60 | validate_return_status=True
```
Comment (petrieh) [inline robottests/ClusterExec.py]: Too long line

File: .pylintrc (hunk 3 above)
Comment (vereszol) [reply .pylintrc]: I'll remove the second occurence

File: robottests/ClusterExec.py (hunk 4 above)
Comment (vereszol) [reply robottests/ClusterExec.py]: I'll remove last 2 arguments from the example

**Comment by vereszol** [comment]
> Please create a new bug correction release out of this by increasing version and describe the change in CHANGES.rst

Done.

File: robottests/test_remoterunner_and_remotescript.robot [hunk 5]
```diff
@@ -27,9 +27,9 @@ Remote Test Setup
     Remove Tmp Directories
 
 Remove Tmp Directories
```
Comment (petrieh) [inline robottests/test_remoterunner_and_remotescript.robot]: END is missing

**Pull request merged**